lib/spack/spack/build_systems/cuda.py

```diff
@@ -93,12 +93,16 @@ def cuda_flags(arch_list):
     conflicts('%clang@7:', when='+cuda ^cuda@10.0.130' + arch_platform)
 
     # Intel is mostly relevant for x86_64 Linux, even though it also
-    # exists for Mac OS X.
-    conflicts('%intel@:14,16:', when='+cuda ^cuda@7.5')
-    conflicts('%intel@:14,17:', when='+cuda ^cuda@8.0.44')
-    conflicts('%intel@:14,18:', when='+cuda ^cuda@8.0.61:9.1')
-    conflicts('%intel@17:18', when='+cuda ^cuda@9.2:')
-    conflicts('%intel@19:', when='+cuda')
+    # exists for Mac OS X. No information prior to CUDA 3.2 or Intel 11.1
+    conflicts('%intel@:11.0', when='+cuda ^cuda@:3.1')
+    conflicts('%intel@:12.0', when='+cuda ^cuda@5.5:')
```

Comment:
according to my [recordings](https://gist.github.com/ax3l/9489132), `intel@12.1` is the only supported version for CUDA 5.5

Reply:
There are a couple of ways to write this. Some more verbose than the others, but also more precise. 

As is currently written, it will allow you to install `intel@12.1` with `cuda@5.5`, but it will also allow `intel@12.2:12.99` to install with `cuda@:5.9`. If these are compiler and CUDA versions that have existed, than as written it is incorrect and I should use the most precision but verbose version below. Though admittedly, we might want the more verbose version just for the sake of clarity.

The alternative, which is much more verbose would read something like this:
`conflicts('%intel@:12.0,12.2:', when='+cuda ^cuda@:5.4,5.6:')`

Do we want the latter (specific and verbose)? If so, we should specify all the conflicts in that file this way.

Reply:
I see, yes the lack of `!@version` is a bit devistating. I am ok with the current form, this seems logical to me. Do you mind adding CUDA 10.0 and 10.1 as well?

Reply:
Yes, I think that I have covered with the next commit. I am not sure that the CUDA package installs 10.1, but that can be handled in a separate PR.

I also filed this github issue for version-to-version specific conflicts.
https://github.com/spack/spack/issues/11007